capsules/src/bme280.rs

```diff
@@ -292,9 +292,13 @@ impl<'a> I2CClient for Bme280<'a> {
                     Operation::None => (),
                     Operation::Temp => {
                         let calib = self.calibration.get();
-                        let adc_temperature = (buffer[0] as usize) << 12
+
+                        // note: per datasheet, measurement is 20-bit two's complement
+                        let adc_temperature: i32 = (((buffer[0] as usize) << 12
                             | (buffer[1] as usize) << 4
-                            | (((buffer[2] as usize) >> 4) & 0x0F);
+                            | (((buffer[2] as usize) >> 4) & 0x0F) << 12)
+                            as i32)
+                            >> 12; // ensure sign extension
```

Comment:
This sign extension breaks the temperature values. Everything else seems to work here

Reply:
Yup, sure does. I missed a parentheses in there, which meant it shifted `buffer[2]` left before or-ing it, per order of operations. Should be fixed in the new commit.